### PR TITLE
MEC Periodic Data matching operation

### PR DESCRIPTION
**Pull request by rahulgudim**
# PR Checklist

Please check if your PR fulfills the following requirements
- [X] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit
)
- [X] Tests for the changes have been added (for bugfixes / features)

# PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [X] Feature (requires Feature flag)
- [ ] Data fix or migration (inert code, no impact until run)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Dependency updates (e.g., add a new gem or update version)

# What is the ticket # detailing the issue?

Ticket: 
https://www.pivotaltracker.com/story/show/185681941

# A brief description of the changes

Current behavior:
There is no Periodic Data Matching happens to fulfill regulatory requirements with State of Maine.

New behavior:
We would like to periodically run a task(twice an year) and want to confirm with the local MEC service that consumers are not eligible for Medicaid, so that regulatory requirements can be fulfilled.

This operation specifically tries to **filter** needed families according to the requirements, send a call to Local MEC service with the `application` payload. 

Followup task
Wrap this operation in a rake task so `DevOps` can conveniently run it when needed.

# Feature Flag

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. A feature flag needs a corresponding environment variable that is used to initialize the state of the flag. Please share the name of the environment variable below that would enable/disable the feature and which client(s) it applies to.

Variable name:

- [ ] DC
- [ ] ME

# Additional Context
Include any additional context that may be relevant to the peer review process.

### CODE REVIEW

File: components/financial_assistance/app/domain/financial_assistance/operations/applications/medicaid_gateway/run_periodic_data_matching.rb [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+          private
+
+          def validate(params)
+            errors = []
+            errors << 'assistance_year param missing' if params[:assistance_year].blank?
+            assistance_year = params[:assistance_year]
+            errors << 'assistance_year param is invalid' unless assistance_year.to_i.to_s == assistance_year.to_s
+            errors << 'transmittable_job_id param is missing' if params[:transmittable_job_id].blank?
+
```
Comment (kristinmerbach) [inline components/financial_assistance/app/domain/financial_assistance/operations/applications/medicaid_gateway/run_periodic_data_matching.rb:33]: add check if batch size optional param given, it should be an integer

**Review by kristinmerbach — COMMENTED** [review]
A couple things to add, but so far looks good!

File: components/financial_assistance/app/domain/financial_assistance/operations/applications/medicaid_gateway/run_periodic_data_matching.rb [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            (0..total_families_count).step(batch_size) do |offset|
+              batch = families.skip(offset).limit(batch_size)
+              batch.each do |family|
+                enrollments = family.hbx_enrollments.by_health.enrolled_and_renewal
+                # Process each record in the batch
+                next unless enrollments.any? {|enrollment| enrollment.applied_aptc_amount.cents.positive? || %w[02 03 04 05 06].include?(enrollment.product.csr_variant_id) }
+                determined_application = fetch_application(family, params[:assistance_year])
+                next unless determined_application.present?
+                total_applications_ran += 1
+                ::FinancialAssistance::Operations::Applications::MedicaidGateway::RequestMecChecks.new.call(application_id: determined_application.id) unless params[:skip_mec_call]
```
[the remark this answers is not in the document]
Comment (rahulgudim) [reply components/financial_assistance/app/domain/financial_assistance/operations/applications/medicaid_gateway/run_periodic_data_matching.rb]: Good point! Added.

**Review by kristinmerbach — APPROVED** [review]
LGTM!

**Review by ymhari — APPROVED** [review]

**Pull request merged**